### ISSUE TITLE
Add Airdrop contract

### ISSUE THREAD

**Issue by demibrener** [OpenZeppelin/zeppelin-solidity#704]
- [ ] 🐛 This is a bug report.
- [X] 📈 This is a feature request.

Introduce an Airdrop smart contract standard to the OpenZeppelin library.

Airdrops became a way for projects to distribute tokens to their community, in an effort to bootstrap the usage and support of their platform. During 2017, we've seen Airdrops performed by projects like OmiseGo and Kyber, among others:
https://github.com/omisego/airdrop
https://github.com/KyberNetwork/airdrop
Also, products are emerging for projects to perform Airdrops: https://news.earn.com/launch-your-token-with-an-earn-com-airdrop-97715970f8c7

With an increasing number of emerging projects in the space struggling to get the adoption they need to justify their market prices, I believe Airdrops will become more popular during 2018. Therefore, it would be great to have an Airdrop smart contract standard in OpenZeppelin, the most popular smart contract library in the space today.

**Comment by frangio** [comment]
Thanks @demibrener! I like the motivation for this feature.

If I understand correctly, an "airdropper" would basically be a "multisend" function.

**Comment by demibrener** [comment]
Thanks @frangio! I believe so.

**Comment by spalladino** [comment]
Note that most of the work in an airdrop are the off-chain tools for:
- collecting the receiving addresses from a blockchain scan (or other criteria)
- automating a huge number of batched transfers

The contract itself should not be that complex, it's just a `transferMany` method, tailored to use as little gas as possible.

**Comment by cedricwalter** [comment]
i wrote an open source airdrop tool using nodejs, require users to registers in a form, their email and phone number, the code is heavily asynchronous and can handle lots of users.

**Comment by spalladino** [comment]
Nice! Care to share a link?

**Comment by cedricwalter** [comment]
https://github.com/StutzCoin/airdrop

**Comment by FradSer** [comment]
@cedricwalter Thank you.

**Comment by subbu-nuco** [comment]
Is anybody working on this ? I would like to work on if nobody is else working on this ?

**Comment by subramanianv** [comment]
Is anybody working on this ? If not, I would like to send a PR for this feature

**Comment by spalladino** [comment]
Not that I know of, feel free to give it a shot!

**Comment by frangio** [comment]
To keep in mind: https://blog.ricmoo.com/merkle-air-drops-e6406945584d

**Comment by xgamermonster9** [comment]
Yes, David, there’s some kind of giveaway going on. I got some Ethereum there yesterday for different wallets. It’s unlikely they’ll take it back from me, but it’s a pity I didn’t have more wallets and couldn’t get more. But maybe it’s for the best.

![Image](https://github.com/user-attachments/assets/ef1112b7-a92e-4f75-9a4c-ac06c20eedb5)

![Image](https://github.com/user-attachments/assets/69b9e50b-dbb9-4bca-a84e-6256355d4441)
